mindmaps-engine/src/main/java/io/mindmaps/engine/controller/VisualiserController.java

```diff
@@ -105,11 +109,11 @@ private String matchQuery(Request req, Response res) {
             final JSONArray halArray = new JSONArray();
 
             withGraph(graph).parseMatch(req.queryParams(REST.Request.QUERY_FIELD))
-                    .stream()
+                    .parallelStream()
```

Comment:
This works with a parallelStream ? I would expect this to throw a closed transaction error since you accessing concepts in different threads. If this is not throwing a closed transaction error then maybe check the number underlaying TitanTransactions being opened here. If this is opening a transaction per concept this could (over time) become a heavy operation I think ?